compiler/front_end/module_ir.py

```diff
@@ -1166,7 +1171,10 @@ def _enum_value_body(indent, docs, attributes, dedent):
 def _external(external, name, colon, comment, newline, external_body):
   del colon, comment, newline  # Unused.
   ir_data_utils.builder(external_body.source_location).start.CopyFrom(external.source_location.start)
-  external_body.name.CopyFrom(name)
+  if external_body.name:
+    ir_data_utils.update(external_body.name, name)
+  else:
+    external_body.name = ir_data_utils.copy(name)
```

Comment:
Why not use `external_body.name = ir_data_utils.copy(name)` in both cases? Or does `CopyFrom()` not overwrite all fields?

Reply:
`update` updates the fields of an *existing* `ir_data` A with the set fields of another `ir_data` B (so fields that are already set in A will not be overwritten if they're not set in B) . In the else case there is no `ir_data` to update, so we create one with `copy`.

Reply:
Yeah, I got the logic in this snippet but `update` delegates to `CopyFrom`. I guessed `CopyFrom` might overwrite existing field but it sounds like that's not the case so we need to keep the existing `external_body.name` around.